src/v/config/node_config.cc

```diff
@@ -26,108 +26,114 @@ node_config::node_config() noexcept
   , data_directory(
       *this,
       "data_directory",
-      "Place where redpanda will keep the data",
+      "Path to the directory for storing Redpanda's streaming data files.",
       {.required = required::yes, .visibility = visibility::user})
   , node_id(
       *this,
       "node_id",
-      "Unique id identifying a node in the cluster. If missing, a unique id "
-      "will be assigned for this node when it joins the cluster",
+      "A number that uniquely identifies the broker within the cluster. If `null` (the default value), Redpanda automatically assigns an ID. If set, it must be non-negative value.
+ The `node_id` property must not be changed after a broker joins the cluster.",
       {.visibility = visibility::user},
       std::nullopt,
       [](std::optional<model::node_id> id) -> std::optional<ss::sstring> {
-          if (id && (*id)() < 0) {
-              return fmt::format("Negative node_id ({}) not allowed", *id);
-          }
-          return std::nullopt;
+    if (id && (*id)() < 0) {
+        return fmt::format("Negative node_id ({}) not allowed", *id);
+    }
+    return std::nullopt;
       })
   , rack(
       *this,
       "rack",
-      "Rack identifier",
+      "A label that identifies a failure zone. Apply the same label to all brokers in the same failure zone. When `enable_rack_awareness` is set to `true` at the cluster level, the system uses the rack labels to spread partition replicas across different failure zones.",
       {.visibility = visibility::user},
       std::nullopt)
   , seed_servers(
       *this,
       "seed_servers",
-      "List of the seed servers used to join current cluster. If the "
-      "seed_server list is empty the node will be a cluster root and it will "
-      "form a new cluster",
+      "List of the seed servers used to join current cluster. If the `seed_servers` list is empty the node will be a cluster root and it will form a new cluster.
```

Comment:
Unfortunately, multi-line rich formatting like this will cause UX issues because we do output these strings in certain scenarios where we expect the string to be a one-line string.  

@dotnwat and I had an idea that we could extend the C++ to have a   _desc   and a _longdesc   member so that we could safely integrate really nice summaries like this into the product.   

summary for today:  any changes coming into this file need to be one-line strings.  We'll need to follow up to accept larger blobs like this one.  I'm happy to get on a call to help clarify what needs to change.